tests/std/tests/P0896R4_ranges_alg_move/test.cpp

```diff
@@ -9,75 +9,69 @@
 
 #include <range_algorithm_support.hpp>
 
+using namespace std;
+
 struct int_wrapper {
     int val                 = 10;
     constexpr int_wrapper() = default;
     constexpr int_wrapper(int x) : val{x} {}
+    int_wrapper(const int_wrapper&) = default;
+    int_wrapper& operator=(const int_wrapper&) = default;
     constexpr int_wrapper(int_wrapper&& that) : val{std::exchange(that.val, -1)} {}
     constexpr int_wrapper& operator=(int_wrapper&& that) {
         val = std::exchange(that.val, -1);
         return *this;
     }
+    auto operator<=>(const int_wrapper&) const = default;
 };
 
-constexpr void smoke_test() {
-    using ranges::move, ranges::move_result, ranges::iterator_t;
-    using std::same_as;
+// Validate that move_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::move_result<int, double>, ranges::in_out_result<int, double>>);
 
-    // Validate that move_result aliases in_out_result
-    STATIC_ASSERT(same_as<move_result<int, double>, ranges::in_out_result<int, double>>);
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::move(borrowed<false>{}, static_cast<int*>(nullptr))),
+    ranges::move_result<ranges::dangling, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::move(borrowed<true>{}, static_cast<int*>(nullptr))), ranges::move_result<int*, int*>>);
 
-    // Validate dangling story
-    STATIC_ASSERT(
-        same_as<decltype(move(borrowed<false>{}, static_cast<int*>(nullptr))), move_result<ranges::dangling, int*>>);
-    STATIC_ASSERT(same_as<decltype(move(borrowed<true>{}, static_cast<int*>(nullptr))), move_result<int*, int*>>);
+struct instantiator {
+    static constexpr int_wrapper expected_output[3] = {13, 55, 12345};
+    static constexpr int_wrapper expected_input[3]  = {-1, -1, -1};
 
-    int const input[] = {13, 53, 12435};
-    {
-        int output[] = {-2, -2, -2};
-        auto result  = move(basic_borrowed_range{input}, basic_borrowed_range{output}.begin());
-        STATIC_ASSERT(same_as<decltype(result),
-            move_result<iterator_t<basic_borrowed_range<int const>>, iterator_t<basic_borrowed_range<int>>>>);
-        assert(result.in == basic_borrowed_range{input}.end());
-        assert(result.out == basic_borrowed_range{output}.end());
-        assert(ranges::equal(output, input));
+    static constexpr void eq(int_wrapper const (&output)[3], int_wrapper const (&input)[3]) {
+        // Extracted into a separate function to keep /analyze from blowing the compiler heap

```

Comment:
Let's rephrase this:
```suggestion
        // Extracted into a separate function to keep /analyze from exhausting the compiler heap
```